Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1190,11 +1190,16 @@ private extension ChatChannelController {
                 switch result {
                 case let .success(payload):
                     self.state = .remoteDataFetched
+
                     self.updateNewestFetchedMessageId(with: payload)
                     self.updateOldestFetchedMessageId(with: payload)
-                    if let pageSize = self.channelQuery.pagination?.pageSize {
-                        self.hasLoadedAllPreviousMessages = payload.messages.count < pageSize
+
+                    let pageSize = self.channelQuery.pagination?.pageSize ?? .messagesPageSize
```

Comment:
Would be nice to add comments what's happening here.

Reply:
Added a comment 👍